primitives/runtime/src/offchain/storage.rs

```diff
@@ -28,6 +28,35 @@ pub struct StorageValueRef<'a> {
 	kind: StorageKind,
 }
 
+/// Result of asking for a particular key from the store.
+#[derive(Debug, PartialEq, Eq)]
+pub enum StorageEntry<T> {
+	/// Successfully retrieved the stored value.
+	Occupied(T),
+	/// Successfully looked up the key but noting is currently stored at that location.
+	Vacant,
+	/// There is no entry in the db for that key.
+	KeyNotStored,
```

Comment:
Very much a low-priority bikeshed, but maybe:

```suggestion
	NoKey,
```